ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -4158,7 +4158,7 @@ public static void addDependencyJars(Configuration conf, Class<?>... classes)
       if (!localFs.exists(new Path(path))) {
         throw new RuntimeException("Could not validate jar file " + path + " for class " + clazz);
       }
-      jars.add(path);
+      jars.add(localFs.makeQualified(new Path(path)).toString());
```

Comment:
should we use the FileUtils helper method?
````
FileUtils.makeQualified(path, conf).toString()
````

Reply:
No, I don't think that will work. It will make it qualified with respect to the FileSystem specified in `Fs.defaultFs` and will lead to the same problem. We need to make it qualified with respect to the `LocalFileSystem`
From the Javadoc
```
   * @return path qualified relative to default file system
   */
  public static Path makeQualified(Path path, Configuration conf) throws IOException {
```